tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -291,95 +259,14 @@ public T data() {
    * @return the tensor raw data mapped to a read-only byte buffer
    * @throws IllegalStateException if the tensor has been closed
    */
-  public ByteDataBuffer rawData() {
-    return TensorBuffers.toBytes(nativeHandle(), true);
-  }
-
-  /** Returns a string describing the type and shape of the Tensor. */
-  @Override
-  public String toString() {
-    return String.format("%s tensor with shape %s", dtype.toString(), shape);
-  }
-
-  /**
-   * Create a Tensor object from a handle to the C TF_Tensor object.
-   *
-   * <p>Takes ownership of the handle.
-   */
-  static Tensor<?> fromHandle(TF_Tensor handle) {
-    Tensor<?> t = new Tensor<>(DataTypes.fromNativeCode(dtype(handle)), Shape.of(shape(handle)));
-    try (PointerScope scope = new PointerScope()) {
-        scope.attach(handle);
-        t.tensorHandle = handle;
-        t.tensorScope = scope.extend();
-    }
-    return t;
-  }
-
-  /**
-   * Create an eager Tensor object from a handle to the C TF_Tensor object.
-   *
-   * <p>Takes ownership of the handle.
-   */
-  static Tensor<?> fromHandle(TF_Tensor handle, EagerSession session) {
-    Tensor<?> t = fromHandle(handle);
-    session.attach(handle);
-    t.tensorScope.detach(handle);
-    return t;
-  }
+  ByteDataBuffer rawData();
 
   /**
    * @return native handle to this tensor
    * @throws IllegalStateException if tensor has been closed
    */
-  TF_Tensor nativeHandle() {
-    return requireHandle(tensorHandle);
-  }
-
-  private PointerScope tensorScope;
-  private TF_Tensor tensorHandle;
-
-  private static TF_Tensor requireHandle(TF_Tensor handle) {
-    if (handle == null || handle.isNull()) {
-      throw new IllegalStateException("close() was called on the Tensor");
-    }
-    return handle;
-  }
-
-  private static TF_Tensor allocate(int dtype, long[] shape, long byteSize) {
-    TF_Tensor t = TF_Tensor.allocateTensor(dtype, shape, byteSize);
-    if (t == null || t.isNull()) {
-      throw new IllegalStateException("unable to allocate memory for the Tensor");
-    }
-    return t;
-  }
-
-  private static int dtype(TF_Tensor handle) {
-    requireHandle(handle);
-    return TF_TensorType(handle);
-  }
-
-  private static long[] shape(TF_Tensor handle) {
-    requireHandle(handle);
-    int numDims = TF_NumDims(handle);
-    long[] dims = new long[numDims];
-    for (int i = 0; i < numDims; ++i) {
-      dims[i] = TF_Dim(handle, i);
-    }
-    return dims;
-  }
+  TF_Tensor nativeHandle();
```

Comment:
This is a public method now right? That's kinda unpleasant.

Reply:
I totally agree. But I did not found (yet) a way to avoid it.